mantle/cmd/kola/testiso.go

```diff
@@ -713,7 +730,14 @@ func testLiveIso(ctx context.Context, inst platform.Install, outdir string, mini
 	if addNmKeyfile {
 		liveConfig.AddSystemdUnit("coreos-test-nm-keyfile.service", verifyNmKeyfile, conf.Enable)
 		targetConfig.AddSystemdUnit("coreos-test-nm-keyfile.service", verifyNmKeyfile, conf.Enable)
+		// NM keyfile via `iso network embed`
 		inst.NmKeyfiles[nmConnectionFile] = nmConnection
+		// nmstate config via live Ignition config, propagated via
+		// --copy-network, which is enabled by inst.NmKeyfiles
+		liveConfig.AddFile(nmstateConfigFile, nmstateConfig, 0644)
+		// XXX - nmstate installs itself in multi-user.target and
+		// there's nothing to run it from coreos-installer.target
+		liveConfig.AddSystemdUnitDropin("NetworkManager.service", "test.conf", "[Unit]\nWants=nmstate.service\n")
```

Comment:
This works around the lack of a `WantedBy=NetworkManager.service` as pointed out by @cgwalters in https://github.com/coreos/fedora-coreos-config/pull/2269#discussion_r1143732123.  `nmstate.service` currently only installs itself in `multi-user.target`, but when we're running coreos-installer via the `coreos.inst.dest_device` kernel argument, we boot to `coreos-installer.target` and don't run `multi-user.target` at all.  We could explicitly add an nmstate dependency for that specific case, but it seems cleaner for `nmstate.service` to automatically start itself whenever NM runs.

Reply:
Fixed upstream in https://github.com/nmstate/nmstate/pull/2285.